apps/03TicTacToe.js

```diff
@@ -24,23 +24,68 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn)
+    ||(board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn)
+    ||(board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn))
+   { 
+      return true;
+   }
+  else 
+  {
+    return false;
+  }
 }
 
 function verticalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn)
+    ||(board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn)
+    ||(board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn))
+   
+  { 
+      return true;
+   }
+  else 
+  {
+    return false;
+  }
 }
 
 function diagonalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn)
+    ||(board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn))
+    { 
+      return true;
+   }
+  else 
+  {
+    return false;
+  }  
 }
 
 function checkForWin() {
-  // Your code here
+  if
+  (horizontalWin() || verticalWin() || diagonalWin())
+{
+  console.log('Player ' + playerTurn + ' Won!');
+  return true;
+}
+else
+{
+return false;
+}
 }
 
 function ticTacToe(row, column) {
-  // Your code here
+  if (playerTurn === 'X'){
+      board[row][column] = 'X';
+      checkForWin();
+      playerTurn = 'O';
+    } 
+    else {
```

Comment:
You can use the value of playerTurn to appropriately assign either 'X' or 'O' to the board.

This will save you from having to write an `if\else` statement.